src/main/java/com/solace/samples/java/patterns/DirectReplierBlocking.java

```diff
@@ -0,0 +1,143 @@
+    private static volatile boolean hasDetectedDiscard = false;  // detected any discards yet?
+    private static volatile boolean isShutdown = false;          // are we done yet?
+
+    public static void main(String... args) throws IOException {
+
+        //1. Make sure that you have all the connection parameters.
+        if (args.length < 3) {  // Check command line arguments
+            System.out.printf("Usage: %s <host:port> <message-vpn> <client-username> [password]%n%n", SAMPLE_NAME);
+            System.exit(-1);
+        }
+        System.out.println(API + " " + SAMPLE_NAME + " initializing...");
+
+        //2. Set up the properties including username, password, vpnHostUrl and other control parameters.
+        final Properties properties = new Properties();
+        setupPropertiesForConnection(properties, args);
+
+        //3. Create the MessagingService object and establishes the connection with the Solace event broker
+        final MessagingService messagingService = com.solace.messaging.MessagingService.builder(ConfigurationProfile.V1)
+                .fromProperties(properties)
+                .build();
+        messagingService.connect();  // blocking connect to the broker
+
+        //4. Register event handlers and callbacks for connection error handling.
+        setupConnectivityHandlingInMessagingService(messagingService);
+
+        //5. Build and start the Receiver object
+        RequestReplyMessageReceiver requestReplyMessageReceiver = messagingService.requestReply().
+                createRequestReplyMessageReceiverBuilder().build(TopicSubscription.of(TOPIC_PREFIX + "*/direct/request/>"));
+        requestReplyMessageReceiver.start();
+        //5-A. Setup an event handler for situations where the reply message could not be published.
+        requestReplyMessageReceiver.setReplyFailureListener(failedReceiveEvent -> System.out.println("### FAILED RECEIVE EVENT " + failedReceiveEvent));
+
+        //6. Create an OutboundMessageBuilder for building the outbound reply message
+        final OutboundMessageBuilder outboundMessageBuilder = messagingService.messageBuilder();
+
+        //7. Define the handler for the incoming message.
+        final RequestReplyMessageReceiver.RequestMessageHandler messageHandler = (inboundMessage, replier) -> {
+
+            msgRecvCounter++;
+
+            //7-A. Since Direct messages, check if there have been any lost any messages
+            checkForDiscardedMessages(inboundMessage);
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+//            System.out.println("The inbound message is : " + inboundMessage.dump()); // Enable this for learning purposes as it logs a String representation of the whole Message
+            System.out.println("The inbound message payload is : " + inboundMessage.getPayloadAsString());
```

Comment:
I believe this will fail if the payload is not string. You can test it out by sending a message using the broker management console and you can specify the type of message as text or bytes.

One way to avoid this is to check the type of the payload and use the appropriate method to extract the payload. For example, this is how it's done in python
https://github.com/SolaceSamples/solace-samples-python/blob/master/patterns/direct_subscriber.py#L18-L22
and this is how we do it in go
https://github.com/SolaceSamples/solace-samples-go/blob/main/patterns/direct_subscriber.go#L18-L25

You can also check out @aaron-613's comment on this community post 
https://solace.community/discussion/comment/5259#Comment_5259

Reply:
Hey @TamimiGitHub  : nice catch. I forgot the point that our payload are transmitted and processed as bytes and we dont have any restriction on the payload that can be published to a topic. 
But I found a nice api method which helps in this.

[InboundMessage. getAndConvertPayload](https://docs.solace.com/API-Developer-Online-Ref-Documentation/pubsubplus-java/com/solace/messaging/receiver/InboundMessage.html#getAndConvertPayload-com.solace.messaging.util.Converter.BytesToObject-java.lang.Class-) which we can use to supply a converter and this is responsible for converting from the byte[] to any format that you want to work with. In this case I have stuck to String but made a comment on its flexibility.

`final String stringPayload = inboundMessage.getAndConvertPayload(
                    (byte[] bytes) -> {
                        return new String(bytes);
                    }, String.class
            );`

Hope this fixes and addressess your point.

Reply:
ouuuh was not aware of [getAndConvertPayload](https://docs.solace.com/API-Developer-Online-Ref-Documentation/pubsubplus-java/com/solace/messaging/receiver/InboundMessage.html#getAndConvertPayload-com.solace.messaging.util.Converter.BytesToObject-java.lang.Class-)

Looks like this will be a good alternative. @aaron-613 is there something similar in JCSMP?